src/main/kotlin/uk/gov/justice/digital/hmpps/externalusersapi/service/UserRoleService.kt

```diff
@@ -74,19 +78,29 @@ class UserRoleService(
   ) {
     // already checked that user exists
     userRepository.findById(userId)?.let { user: User ->
-      maintainUserCheck.ensureUserLoggedInUserRelationship(user.name)
+
+      // MaintainImsUser doesn't need to check user relationship as is system role
+      if (!authenticationFacade.isMaintainImsUser()) {
+        // check that the logged in user has permission to modify user
+        maintainUserCheck.ensureUserLoggedInUserRelationship(user.name)
+      }
+
       val formattedRoles = roleCodes.map { formatRole(it) }
-      val allAssignableRoles = getAllAssignableRolesByUserId(userId)
       for (roleCode in formattedRoles) {
         // check that role exists
         val role = roleRepository.findByRoleCode(roleCode) ?: throw UserRoleException("role", "role.notfound")
 
-        val userRoles = getUserRoles(userId) ?: throw NotFoundException("usernotfound")
+        val userRoles = roleRepository.findRolesByUserId(userId).toList()
```

Comment:
getUserRoles was doing another maintainUserCheck which is not needed so just getting the user roles now